Sources/Valet/SecureEnclaveValet.swift

```diff
@@ -295,3 +400,97 @@ extension SecureEnclaveValet {
     }
 
 }
+
+// MARK: - Private Extensions
+
+extension SecureEnclaveAccessControl {
+
+    /// The `LAPolicy` that the `SecureEnclaveAccessControl` corresponds to.
+    #if !os(watchOS)
+    @available(tvOS 10.0, *)
+    fileprivate var policy: LAPolicy {
+        switch self {
+        case .userPresence, .devicePasscode:
+            return .deviceOwnerAuthentication
+        case .biometricAny, .biometricCurrentSet:
+            if #available(macOSApplicationExtension 10.12.2, *) {
+                return .deviceOwnerAuthenticationWithBiometrics
+            } else {
+                return .deviceOwnerAuthentication
+            }
+        }
+    }
+    #endif
+
+}
+
+// MARK: - Internal Types
+
+/// A type that statically takes an `LAError` and a `SecureEnclaveAccessControl`
+/// and transforms it to either a `KeychainError` or `SecureEnclaveError`
+enum LAErrorTransformer {
+
+    enum TransformedError: Equatable {
+        case keychain(KeychainError)
+        case secureEnclave(SecureEnclaveError)
+    }
+
+    #if !os(watchOS)
+    @available(tvOS 10.0, *)
+    static func transform(
+        error: LAError,
+        accessControl: SecureEnclaveAccessControl
+    ) -> TransformedError {
+        switch (error.code, accessControl) {
+        case (.userCancel, _),
+             (.systemCancel, _),
+             (.authenticationFailed, _),
+             (.appCancel, _):
+            return .secureEnclave(.userCancelled)
```

Comment:
Can we use `.keychain(. userCancelled)` here and delete `.secureEnclave(.userCancelled)`?